src/invidious/routes/channels.cr

```diff
@@ -187,6 +192,38 @@ module Invidious::Routes::Channels
     templated "community"
   end
 
+  def self.post(env)
+    # /post/{postId}
+    id = env.params.url["id"]
+    ucid = env.params.query["ucid"]?
+
+    prefs = env.get("preferences").as(Preferences)
+
+    locale = prefs.locale
+    region = env.params.query["region"]? || prefs.region
+
+    thin_mode = env.params.query["thin_mode"]? || prefs.thin_mode
+    thin_mode = thin_mode == "true"
+
+    client_config = YoutubeAPI::ClientConfig.new(region: region)
+
+    if !ucid.nil?
+      ucid = ucid.to_s
+      post_response = fetch_channel_community_post(ucid, id, locale, "json", thin_mode)
+    else
+      # resolve the url to get the author's UCID
+      response = YoutubeAPI.resolve_url("https://www.youtube.com/post/#{id}")
+      return error_template(400, "Invalid post ID") if response["error"]?
+
+      ucid = response.dig("endpoint", "browseEndpoint", "browseId").as_s
+      params = response.dig("endpoint", "browseEndpoint", "params").as_s
```

Comment:
I'm wondering: what are contained in this params that we might need?
Can we craft some of that somehow?

Reply:
I should be able to craft it if that's preferred

Reply:
if it's not too big nor too complex, yeah, I'd prefer if we have control over the data!

Reply:
I actually already craft these params when we don't have to resolve the url so it would actually simplify my code 😅